autoPyTorch/api/base_task.py

```diff
@@ -143,40 +744,100 @@ def predict(
         Returns:
             Array with estimator predictions.
         """
-        # TODO use the batch size and take it from the pipeline
-        # TODO argument to method for a flag that might return probabilities
-        # in case the pipeline does not have a predict_proba then continue
-        # normally and raise warning
-        return self.pipeline.predict(X_test)
+
+        # Parallelize predictions across models with n_jobs processes.
+        # Each process computes predictions in chunks of batch_size rows.
+        if self._logger is None:
+            self._logger = self._get_logger("Predict-Logger")
+
+        if self.ensemble_ is None and not self._load_models(self.resampling_strategy):
+            raise ValueError("Failed to fit an ensemble. Cannot predict")
```

Comment:
Maybe "No ensemble found. Either fit has not yet been called or no ensemble was fitted"